README.md

```diff
@@ -113,13 +113,13 @@ rails s
 To run a single Rspec test file, you can do the following to run the tests in the `your_spec.rb` file:
 
 ```shell
-rspec spec rspec/your_spec.rb
+rspec spec/model/user_spec.rb
 ```
 
 To run all Rspec test files in a directory
 
 ```shell
-rspec spec rspec
+rspec spec/models
 ```
 
 <p align="right">(<a href="#readme-top">back to top</a>)</p>
```

Comment:
The code in your `Entity`, `Category`, and `CategoryEntity` Models seems to implement a one-to-many relationship between `Category` and `Entity` through the `CategoryEntity` join model, rather than a many-to-many relationship as required for this project in the [Entity relationship diagram](https://github.com/microverseinc/curriculum-rails/raw/main/capstone/images/erd_diagram.png). In a many-to-many relationship, you would typically use a `has_and_belongs_to_many` association or a more explicit two `has_many through:` associations.

Let's correct the implementation to achieve a many-to-many relationship between `Category` and `Entity`:

```ruby
class Entity < ApplicationRecord
  belongs_to :author, class_name: 'User'
  has_many :category_entities, dependent: :destroy
  has_many :categories, through: :category_entities

  validates :name, presence: true
  validates :amount, numericality: { greater_than_or_equal_to: 0 }

  def date
    "#{created_at.day}/#{created_at.month}/#{created_at.year}"
  end

  def total
    "$#{amount}"
  end
end

class Category < ApplicationRecord
  belongs_to :author, class_name: 'User'
  has_many :category_entities, dependent: :destroy
  has_many :entities, through: :category_entities

  validates :name, presence: true
  validates :icon, presence: true
  validates :total_amount, numericality: { greater_than_or_equal_to: 0 }

  before_validation :set_default_value

  def date
    "#{created_at.day}/#{created_at.month}/#{created_at.year}"
  end

  def total
    "$#{total_amount}"
  end

  def set_default_value
    self.total_amount ||= 0.0
  end
end

class CategoryEntity < ApplicationRecord
  after_create :increase_category_total_amount

  belongs_to :entity
  belongs_to :category

  private

  def increase_category_total_amount
    category.increment!(:total_amount, entity.amount)
  end
end
```

In this corrected implementation, I am using the `has_many :through` association to establish a many-to-many relationship between `Category` and `Entity` through the `CategoryEntity` join model. The `has_many :through` association allows you to work with the join model explicitly and perform operations on it, as well as retrieve associated records efficiently.

This is an example of what your migration files might look like for each model based on the corrected implementation.

1. Entity Migration:

```ruby
class CreateEntities < ActiveRecord::Migration[7.0]
  def change
    create_table :entities do |t|
      t.references :author, foreign_key: { to_table: 'users' }
      t.string :name
      t.decimal :amount, precision: 10, scale: 2
      t.timestamps
    end
  end
end
```

2. Category Migration:

```ruby
class CreateCategories < ActiveRecord::Migration[7.0]
  def change
    create_table :categories do |t|
      t.references :author, foreign_key: { to_table: 'users' }
      t.string :name
      t.string :icon
      t.decimal :total_amount, precision: 10, scale: 2, default: 0.0
      t.timestamps
    end
  end
end
```

3. CategoryEntity Migration:

```ruby
class CreateCategoryEntities < ActiveRecord::Migration[7.0] 
  def change
    create_table :category_entities do |t|
      t.references :entity, foreign_key: true # This is the most important 
      t.references :category, foreign_key: true # This is the most important 
      t.timestamps
    end
  end
end
```

You may need to drop your Database, delete the affected Migration files, and generate new migration files based on the above changes. After creating these migrations, you can run `rails db:create db:migrate` to apply them to your database.

Please test your application to ensure it works as expected after implementing these changes.